test-data/unit/check-narrowing.test

```diff
@@ -1027,5 +1027,27 @@ if str_or_bool_literal is not True and str_or_bool_literal is not False:
     reveal_type(str_or_bool_literal)  # N: Revealed type is "builtins.str"
 else:
     reveal_type(str_or_bool_literal)  # N: Revealed type is "Union[Literal[False], Literal[True]]"
+[builtins fixtures/primitives.pyi]
+
+[case testNarrowingBoolean]
+# flags: --strict-optional
+from typing import Optional
+from typing_extensions import Literal
+
+bool_val: bool
 
+if bool_val is not False:
```

Comment:
Should `if bool_val` and `if not bool_val` also do narrowing? These can't be overridden for `bool`.

Reply:
Yeah, that seems useful, i'll look into it 👍